### PR TITLE
UGENE-7468. Migrate Stockholm format to the correct Unicode support

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2Formats/src/StockholmFormat.cpp [hunk 1]
```diff
@@ -94,491 +75,331 @@ enum AnnotationTag {
...
-    }
-    virtual ~Annotation() {
-    }
-};
-//#=GF annotations
-struct FileAnnotation : public Annotation {
-    FileAnnotation(AnnotationTag t, QString v)
-        : Annotation(FILE_ANNOTATION, t, v) {
+/** Internal representation of the annotation used internally during parsing. */
+class StockholmAnnotation {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/StockholmFormat.cpp]: Please, do something about thisc class - split declarations/definitions and move to separate files

File: src/corelibs/U2Formats/src/StockholmFormat.cpp [hunk 2]
```diff
@@ -94,491 +75,331 @@ enum AnnotationTag {
...
-            assert(SS_CONS == ann->tag || RF == ann->tag);
-            Annotation *nAnn = findAnnotation(ann_list, COLUMN_ANNOTATION, ann->tag);
-            if (nullptr != nAnn) {
-                nAnn->val.append(ann->val);
-                delete ann;
+/**
+ * A list of StockholmAnnotation with an adjusted 'add' method.
+ * Auto-deletes all annotations when destroyed.
+ */
+class AnnotationBank {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/StockholmFormat.cpp:100]: The same

File: src/corelibs/U2Formats/src/StockholmFormat.cpp [hunk 3]
```diff
@@ -94,491 +75,331 @@ enum AnnotationTag {
...
-    virtual ~Annotation() {
-    }
-};
-//#=GF annotations
-struct FileAnnotation : public Annotation {
-    FileAnnotation(AnnotationTag t, QString v)
-        : Annotation(FILE_ANNOTATION, t, v) {
+/** Internal representation of the annotation used internally during parsing. */
+class StockholmAnnotation {
+public:
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Formats/src/StockholmFormat.cpp]: If it's fully-public class, so, it's deffinetely a structure. I would suggest use this type in this case

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
See comments

File: src/corelibs/U2Formats/src/StockholmFormat.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/StockholmFormat.cpp]: Why? There is nothing bad in local API defined locally. 

All modern languages I know use this style and the archaic C/C++ way of declaration/definition separation is a solution for a different problem: the one we do not have in this case.

File: src/corelibs/U2Formats/src/StockholmFormat.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/StockholmFormat.cpp:100]: The same.

File: src/corelibs/U2Formats/src/StockholmFormat.cpp (hunk 3 above)
Comment (yalgaer) [reply src/corelibs/U2Formats/src/StockholmFormat.cpp]: Fixed: made it a struct.

**Pull request merged**